### ISSUE TITLE
How should version branches be named?

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#3249]
Now that we're using semantic versioning, we'll need the ability to go back and make releases for earlier version lines than what master targets (i.e., patch releases). That means we need separate branches to maintain the multiple heads. The question is, how should these version branches be named?

To provide some background, I'll point out that master will hold the code for the latest version line. That means the latest major + minor (e.g., 2.0.x). All other version lines should be in branches. Each branch holds a minor version line.

One idea is to name the version line branches using a concrete major and minor number and an "x" to represent the next patch number:

```
<major>.<minor>.x (e.g., 2.0.x)
```

Another idea is to leave off the patch segment:

```
<major>.<minor> (e.g., 2.0)
```

(This is consistent with how the version number is specified for the docs).

Then there's the question of whether to include the leading "v" like tags have.

```
v<major>.<minor>.x (e.g., v2.0.x)
```

Whatever we choose, we just need to be consistent in this project, and perhaps across the ecosystem.

**Comment by mojavelinux** [comment]
cc: @Mogztter @graphitefriction

**Comment by Mogztter** [comment]
I would recommend to not include a leading "v".
It helps to distinguish between tags and branches (especially when you use "git checkout v" completion).

Then I'm fine with either `2.0.x` and `2.0` with a slight preference for `2.0.x` where it's clear that the project is using semantic versioning.

**Comment by mojavelinux** [comment]
That's the direction I'm leaning too. I'll check with Sarah to see if she's on board. If so, we'll proceed.

**Comment by mojavelinux** [comment]
As for the branching strategy itself, here are my thoughts:

Changes for the next minor release (or an eminent major release) go to master. The current minor gets siphoned off to a branch. I prefer this model because I don't like the temptation of letting improvements accumulate in a background branch. I think it makes it harder for people to see the activity in the project and thus less likely to participate.

I don't think this branch should be created immediately after release. Instead, it should wait for a cooling off period (a few weeks) during which patch releases are likely. Once the release line has stabilized, then the branch is created. This avoids the problem of having to merge a bunch of times into master when the only activity is bug fixes.

In other words:

* Publish a .0 release
* Make bug fixes to master and publish patch releases
* Create a minor branch and update the version in master to be the next minor (or major if the situation calls for it)
* Commit changes for the next minor (or major) to master
* Repeat

**Comment by mojavelinux** [comment]
> I would recommend to not include a leading "v".

The reason I prefer it is that it makes pattern matching a lot simpler, particularly for Antora. It helps distinguish from non-release-line branches if they ever show up. And we can now see an example of that now with the feature/ branch for the modern HTML converter.

We've been using the `v<major>.<minor>.x` (e.g., v2.0.x) pattern with Antora quite successfully for several years now. And other Asciidoctor projects have now adopted it, such as Asciidoctor.js and Asciidoctor PDF. I think we should stick with what's working.

I do want to follow-up by putting these notes into the CONTRIBUTING guide (for lack of a better place) so it's clear how the branches are managed.

**Comment by mojavelinux** [comment]
I forgot that we already have a page for versioning and release policies in the community docs. I captured this information on that page. https://docs.asciidoctor.org/about/version-and-release-policies/#version-line-branches
